experimental/form_app/lib/src/validation.dart

```diff
@@ -0,0 +1,161 @@
+          ),
+        ],
+      ),
+      body: Form(
+        key: _formKey,
+        child: Scrollbar(
+          child: SingleChildScrollView(
+            padding: EdgeInsets.all(16),
+            child: Column(
+              children: [
+                // A text field that validates that the text is an adjective.
+                TextFormField(
+                  validator: (value) {
+                    if (value.isEmpty) {
+                      return 'Please enter an adjective.';
+                    }
+                    if (english_words.adjectives.contains(value)) {
+                      return null;
+                    }
+                    return 'Not a valid adjective.';
+                  },
+                  decoration: InputDecoration(
+                    filled: true,
+                    hintText: 'e.g. quick, beautiful, interesting',
+                    labelText: 'Enter an adjective',
+                  ),
+                  onChanged: (value) {
+                    adjective = value;
+                  },
+                ),
+                SizedBox(
+                  height: 24,
+                ),
+                // A text field that validates that the text is a noun.
+                TextFormField(
+                  validator: (value) {
+                    if (value.isEmpty) {
+                      return 'Please enter a noun.';
+                    }
+                    if (english_words.nouns.contains(value)) {
+                      return null;
+                    }
+                    return 'Not a valid noun.';
+                  },
+                  decoration: InputDecoration(
+                    filled: true,
+                    hintText: 'i.e. a person, place or thing',
+                    labelText: 'Enter a noun',
+                  ),
+                  onChanged: (value) {
+                    noun = value;
+                  },
+                ),
+                SizedBox(
+                  height: 24,
+                ),
+                // A custom form field that requires the user to check a
+                // checkbox.
+                FormField(
+                  initialValue: false,
+                  validator: (value) {
+                    if (value == false) {
+                      return 'You must agree to the terms of service.';
+                    }
+                    return null;
+                  },
+                  builder: (FormFieldState formFieldState) {
+                    return Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Row(
+                          mainAxisAlignment: MainAxisAlignment.start,
+                          children: [
+                            Checkbox(
+                              value: agreedToTerms,
+                              onChanged: (value) {
+                                // When the value of the checkbox changes, we
```

Comment:
nit: @kwalrath can correct me here (srsly!), but generally I try to avoid first person pronouns in comments like this. My usual cheat is to say, "the app..." in place of "we" or "I."

Reply:
Agreed. You could even drop the subject:

```
// When the value of the checkbox changes, update
// the FormFieldState so the form is revalidated.
```